client/go/outline/vpn/vpn_linux.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vpn
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"net"
+	"sync"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline"
+	perrs "github.com/Jigsaw-Code/outline-apps/client/go/outline/platerrors"
+	gonm "github.com/Wifx/gonetworkmanager/v2"
+)
+
+type linuxVPNConn struct {
```

Comment:
This type is doing a lot. See the other comments. We need to clarify side-effects and some method are better off being standalone.

Reply:
It's also not super clear what this type represents, perhaps some comments will help.

Reply:
Added some comments. This is the Linux implementation of `VPNConnection`. We used Linux-only APIs (which is not compilable on when targeting Windows).

But I may move some common logic (for example, the `WaitGroup`s, and the traffic copying goroutines) to the `vpn.go` file, because these can be shared between Windows and Linux, but still keep the Linux specific calls in this `vpn_linux.go` file. Though this move might be done in the next PR (when introducing the callback, because the callback of status change logic is shared).